core/capabilities/reader.go

```diff
@@ -0,0 +1,104 @@
+package capabilities
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	kcr "github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/keystone_capability_registry"
+	p2ptypes "github.com/smartcontractkit/chainlink/v2/core/services/p2p/types"
+	evmrelaytypes "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+)
+
+type remoteRegistryReader struct {
+	r types.ContractReader
+}
+
+var _ reader = (*remoteRegistryReader)(nil)
```

Comment:
Out of curiosity, why isn't this part of the `syncer.go` file? Is it to keep the file size smaller or some go pattern?

Reply:
Do you mean this type assertion specifically? It's located here because it makes assertions about the reader abstraction and the interface it satisfies. Separating these out also makes it easier to mock the reader so we don't have to set up a chain and make the relevant registry calls.

Reply:
Sorry, I should've been clearer 🙈 I was wondering why the reader.go was split from the syncer.go. After reviewing the whole PR, it might make sense for the reader to do the reading and state augmentation though and the syncer being a simple service that manages the reader 🤷‍♂️

Reply:
Makes sense